src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -210,22 +208,42 @@ private protected (Solution oldSolution, Solution newSolution) SetCurrentSolutio
         /// allowing that memory to be shared.
         /// </summary>
         /// <param name="transformation">Solution transformation.</param>
-        /// <param name="changeKind">The kind of workspace change event to raise.</param>
-        /// <param name="projectId">The id of the project updated by <paramref name="transformation"/> to be passed to
-        /// the workspace change event.</param>
-        /// <param name="documentId">The id of the document updated by <paramref name="transformation"/> to be passed to
-        /// the workspace change event.</param>
+        /// <param name="changeKind">The kind of workspace change event to raise. The id of the project updated by
+        /// <paramref name="transformation"/> to be passed to the workspace change event.  And the id of the document
+        /// updated by <paramref name="transformation"/> to be passed to the workspace change event.</param>
         /// <returns>True if <see cref="CurrentSolution"/> was set to the transformed solution, false if the
         /// transformation did not change the solution.</returns>
         private protected (bool updated, Solution newSolution) SetCurrentSolution(
             Func<Solution, Solution> transformation,
-            Func<Solution, Solution, WorkspaceChangeKind> changeKind,
-            ProjectId? projectId = null,
-            DocumentId? documentId = null,
+            Func<Solution, Solution, (WorkspaceChangeKind changeKind, ProjectId? projectId, DocumentId? documentId)> changeKind,
```

Comment:
I wonder if it makes sense just to have the transformation also return the change info too. But that can be another PR.

Reply:
that's a reasonable idea.  thanks!